client/blocks/taxonomy-manager/list.jsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+import {
+	includes,
+	filter,
+	map,
+	memoize,
+	noop,
+	reduce,
+	union,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import VirtualList from 'components/virtual-list';
+import ListItem from './list-item';
+import CompactCard from 'components/card/compact';
+import { decodeEntities } from 'lib/formatting';
+import QueryTerms from 'components/data/query-terms';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	isRequestingTermsForQueryIgnoringPage,
+	getTermsLastPageForQuery,
+	getTermsForQueryIgnoringPage,
+} from 'state/terms/selectors';
+
+/**
+ * Constants
+ */
+const DEFAULT_TERMS_PER_PAGE = 100;
+const LOAD_OFFSET = 10;
+const ITEM_HEIGHT = 55;
+
+export class TaxonomyManagerList extends Component {
+	static propTypes = {
+		terms: PropTypes.array,
+		taxonomy: PropTypes.string,
+		search: PropTypes.string,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+		lastPage: PropTypes.number,
+		onTermClick: PropTypes.func,
+	};
+
+	static defaultProps = {
+		loading: true,
+		terms: [],
+		onNextPage: noop,
+		onTermClick: noop,
+	};
+
+	state = {
+		requestedPages: [ 1 ]
+	};
+
+	componentWillMount() {
+		this.itemIds = map( this.props.terms, 'ID' );
+		this.getTermChildren = memoize( this.getTermChildren );
```

Comment:
After looking at the implementation, I can say that we don't have to worry about clearing the cache on `componentWillUnmount`, but you're right about the changes to `this.props.terms`. So I have two options, add a `componentWillReceiveProps` and clear the cache there or drop the cache completely. 

I'm opting for the second to "Avoid premature optimization". We could optimize if we feel the need (tested with a large number of categories)